### PR TITLE
unified trace into device layer

### PR DESCRIPTION
**Pull request by hackwa**
first cut tested on hw and hw emulation

### CODE REVIEW

File: src/runtime_src/xdp/profile/device/device_trace_offload.h [hunk 1]
```diff
@@ -40,57 +41,86 @@ enum class DeviceOffloadStatus {
...
+    bool read_trace_init();
+    XDP_EXPORT
+    void read_trace_end();
+
+public:
+    XDP_EXPORT
     void set_trbuf_alloc_sz(uint64_t sz) {
         m_trbuf_alloc_sz = sz;
     };
+    XDP_EXPORT
```
Comment (IshitaGhosh) [inline src/runtime_src/xdp/profile/device/device_trace_offload.h]: Export keyword is not needed for methods declared in header files. Please add the change.

Comment (hackwa) [reply src/runtime_src/xdp/profile/device/device_trace_offload.h]: removed exports

**Comment by abalasa** [comment]
retest this please

**Comment by abalasa** [comment]
retest this please

**Pull request merged**